### ISSUE TITLE
XTandem N-terminal acetylation of peptides vs proteins

### ISSUE THREAD

**Issue by oliverschilling** [OpenMS/OpenMS#3467]
In the X!Tandem adapter (2.3) I chose the  modification  'Acetyl (N-term)' as a fixed modification to identify peptides with acetylated N-termini (introduced as a chemical modification). The adapter gave the following message
```
Acetyl (N-term)' is handled implicitly by the X! Tandem option 'protein, quick acetyl'. Set the 'force' flag in XTandemAdapter to force explicit inclusion of this modification.
``` 
Consequently, peptide-level acetylation was not probed. This seems to be a misinterpretation within the X!Tandem adapter. Note that protein-level acetylation only probes acetylation of the first/second residue of a protein sequence whereas peptide-level acetylation (`42 28.03130@[` in X!Tandem code) probes acetylation of every peptidic N-terminus.

**Comment by lars20070** [comment]
[Here](https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/XTandemInfile.cpp#L353) the relevant code.

**Comment by jpfeuffer** [comment]
Are you sure that peptide N-term is not considered by XTandem? You do not have to force that option.

**Comment by jpfeuffer** [comment]
And that also only applies only for variable modifications (see code) and you said you put it there as a fixed modification.

**Comment by oliverschilling** [comment]
Absolutely sure!

"'protein, quick acetyl’.  looks only for acetylation of the 1st or 2nd residue; this is to probe for co-translational acetylation in eukaryotes. Hence it is NOT equivalent "Acetyl (N-term)” which probes acetylation of every peptide.

True; this was a variable modification; not static.

**Comment by jpfeuffer** [comment]
Yes but "protein, quick acetyl" is only added if you turn on the "force" parameter, as stated by the note that you received by the tool.

**Comment by jpfeuffer** [comment]
If "force" is not set to true, it should in theory only be handled as a normal variable modification.

**Comment by oliverschilling** [comment]
that is not the point. The point is that "Acetyl (N-term)” is different from the X! Tandem option 'protein, quick acetyl’ and there is no reason to consider them equivalent as indicated by the statement in the adapter "Modification 'Acetyl (N-term)' is handled implicitly by the X! Tandem option 'protein, quick acetyl’”

**Comment by jpfeuffer** [comment]
Yes, you can replicate the correct behaviour I think for now. One problem with the misleading messages is that we currently do not support a direct differentiation between peptide and protein terminus in our modifications. That is an open issue.

**Comment by lars20070** [comment]
@jpfeuffer We test for `not default mods` but then set [`default mods = true`](https://www.thegpm.org/TANDEM/api/pqa.html).

**Comment by oliverschilling** [comment]
the “force” workaround works nicely; thank you.

See unimod for the different possibilities of acetylation:
http://www.unimod.org/modifications_view.php?editid1=1 <http://www.unimod.org/modifications_view.php?editid1=1>

In particular, unimod distinguishes between acetylation at a protein N-terminus "Position 	Protein N-term” and any peptide N-terminus "Position 	Any N-term”.

**Comment by lars20070** [comment]
@cbielow Can we pass the [`protein, quick acetyl`](https://www.thegpm.org/TANDEM/api/pqa.html) flag directly to the user interface? Might be less confusing than linking it to the [`force`](https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/XTandemInfile.cpp#L353) flag.
